src/mudlet.cpp

```diff
@@ -1803,34 +1747,53 @@ Host* mudlet::getActiveHost()
 {
     if (mpCurrentActiveHost && mpCurrentActiveHost->mpConsole) {
         return mpCurrentActiveHost;
-    } else {
-        return nullptr;
     }
+
+    return nullptr;
 }
 
+// Received when the OS/DE/WM tells Mudlet to close (or we force the close
+// ourselves):
 void mudlet::closeEvent(QCloseEvent* event)
 {
-    QVector<QString> closingHosts;
+    qDebug() << "mudlet::closeEvent(...) INFO - called!";
 
+    QStringList hostsToDestroy;
+    bool abortClose = false;
+    // Due to the way that Hosts are stored we cannot do a closeHost(hostName)
+    // within the following loop as it fatally messes with what mHostManager
+    // contains - this is STL iterator stuff! Slysven 2024/04
```

Comment:
This code bit me - and since it is our only (so far) use of `QScopedPointer<T>` I wanted to warn anyone tempted to frob things here - and since I found it it didn't seem unreasonable to identify myself in case anyone else wanted to ask why this comment was here without having to go git history reading...

Reply:
I don't foresee anyone taking you up on this offer, as the Qt documentation is rather straightforward here. I don't think we should start this trend...